test/linalg.jl

```diff
@@ -31,38 +38,45 @@ using DataFrames, LinearAlgebra, MixedModels, Random, SparseArrays, StatsModels,
 end
 
 @testset "reweight!" begin
-    rng = Random.MersenneTwister(1234321)
-    df = (Y = randn(rng, 400), A = repeat(['N','Y'], outer=200),
-        G = repeat('A':'T', inner = 2, outer=10), H = repeat('a':'j', inner=40))
-    lmm1 = fit!(LinearMixedModel(@formula(Y ~ 1+A+(1+A|G)+(1+A|H)), df,
-        wts  = ones(400)))
-    @test loglikelihood(lmm1) ≈ -578.9080978272708
-    MixedModels.reweight!(lmm1, ones(400))
-    @test loglikelihood(fit!(lmm1)) ≈ -578.9080978272708
+    rng = MersenneTwister(1234321)
+    df = (
+        Y = randn(rng, 400),
+        A = repeat(PooledArray(["N","Y"]), outer=200),
+        G = repeat(PooledArray(string.('A':'T')), inner = 2, outer=10),
+        H = repeat(PooledArray(string.('a':'j')), inner=40),
+        )
+    m1 = fit(MixedModel, @formula(Y ~ 1 + A + (1+A|G) + (1+A|H)), df)
+    wm1 = fit(MixedModel, @formula(Y ~ 1+A+(1+A|G)+(1+A|H)), df, wts  = ones(400))
+    @test loglikelihood(wm1) ≈ loglikelihood(m1)
```

Comment:
is this necessary? or could you just deepcopy `m1`, the do `reweight!` and check the fits? (Or do we need weights in the original fit to able to re-weight?)